src/Microsoft.PowerApps.TestEngine/PowerApps/PowerAppFunctions.cs

```diff
@@ -275,8 +276,13 @@ public async Task<bool> SetPropertyDateAsync(ItemPath itemPath, DateValue value)
                 var propertyNameString = JsonConvert.SerializeObject(itemPath.PropertyName);
                 var recordValue = value.GetConvertedValue(null);
 
-                // Date.parse() parses the date to unix timestamp
-                var expression = $"PowerAppsTestEngine.setPropertyValue({itemPathString},{{{propertyNameString}:Date.parse(\"{recordValue}\")}})";
+                // Date.parse() parses the date to unix timestamp.
+                // SetProperty value expected from user is Date(yyyy, MM, dd)
+                // Setting the time value to T00:00.000Z to maintain uniformity across timezones.
+                // This value explicitly specifies the UTC timezone. 
+                // This helps in fetching the date value as it is in any timezone locally without considering one-off in day value.
+                var dt = $"Date.parse(\"{recordValue.Date.ToString("yyyy-MM-dd")}{UTCTimeValue}\")";
```

Comment:
Should this be handled in the JSSDK?

Reply:
We could potentially change it there too. I thought since we have an option to pass the right value here much earlier in the lifecycle(we have separate methods for each variable type , i.e date, record, table etc), it is easier. Are you concerned about how secure this string is ?

Reply:
I'm not worried about the security since there isn't any new user-provided strings. I'm just curious about the trade-offs between forcing a user to update the Test Engine versus having to republish their app to get the fix. Just wanting to make sure you've considered the implications of each of these options & that this is the correct one... (not sure if there was an Analyze on this, if so, I apologize if this was discussed already...)

Reply:
Sure, good point. Considering the easiest option for users. Here since the changes are in the TestEngine code and not in jssdk, user would have to just get the latest TE.